src/abstract-interpretation/data-frame/mappers/function-mapper.ts

```diff
@@ -141,13 +157,110 @@ function mapDataFrameConvert(
+			const sources = findSource(fileName, { referenceChain: referenceChain });
+
+			if(sources?.length === 1) {
+				source = sources[0];
+				request = getSourceProvider().createRequest(source);  // create request from resolved source file path
+			} else if(params.text === undefined && unescapeSpecialChars(fileName).includes('\n')) {
+				request = requestFromInput(unescapeSpecialChars(fileName));  // create request from string if file name argument contains newline
+			}
+		}
+	} else if(textArg !== undefined && textArg !== EmptyArgument) {
+		const text = resolveIdToArgValue(textArg, info);
+
+		if(typeof text === 'string') {
+			source = text;
+			request = requestFromInput(unescapeSpecialChars(text));
+		}
+	}
+	if(request === undefined) {
+		return [{
+			operation: 'read',
+			operand:   undefined,
+			args:      { source: source, colnames: undefined, rows: undefined }
+		}];
+	}
+	const headerArg = getFunctionArgument(args, params.header, info);
+	const separatorArg = getFunctionArgument(args, params.separator, info);
+	const quoteArg = getFunctionArgument(args, params.quote, info);
+	const commentArg = getFunctionArgument(args, params.comment, info);
+	const skipLinesArg = getFunctionArgument(args, params.skipLines, info);
+	const header = headerArg ? resolveIdToArgValue(headerArg, info) : params.header.default;
+	const separator = separatorArg ? resolveIdToArgValue(separatorArg, info) : params.separator.default;
+	const quote = quoteArg ? resolveIdToArgValue(quoteArg, info) : params.quote.default;
+	const comment = commentArg ? resolveIdToArgValue(commentArg, info) : params.comment.default;
+	const skipLines = skipLinesArg ? resolveIdToArgValue(skipLinesArg, info) : params.skipLines.default;
+
+	if(typeof header !== 'boolean' || typeof separator !== 'string' || typeof quote !== 'string' || typeof comment !== 'string' || typeof skipLines !== 'number') {
+		return [{
+			operation: 'read',
+			operand:   undefined,
+			args:      { source: source, colnames: undefined, rows: undefined }
+		}];
+	}
+	const LineCommentRegex = new RegExp(`\\s*[${comment}].*`);
+	let firstLine = undefined as (string | undefined)[] | undefined;
+	let firstLineNumber = 0;
+	let rowCount = 0;
+	let allLines = true;
+
+	const parseLine = (line: Buffer | string, lineNumber: number) => {
+		const text = comment ? line.toString().replace(LineCommentRegex, '') : line.toString();
+
+		if(text.length > 0 && lineNumber >= (skipLines ?? 0)) {
+			if(firstLine === undefined) {
+				firstLine = getEntriesFromCsvLine(text, separator, quote, comment);
+				firstLineNumber = lineNumber;
+			}
+			if(!header || lineNumber > firstLineNumber) {
+				rowCount++;
+			}
+		}
+	};
+
+	if(request.request === 'text') {
+		request.content.split('\n').forEach(parseLine);
+	} else if(request.request === 'file') {
+		allLines = readLineByLineSync(request.content, parseLine, MaxReadLines);
+	}
```

Comment:
I would again outsource this into a function that acecpts a parser for content consumptions